tests/modules/thunder_test_mock_request/src/ThunderTestMockRequestServiceProvider.php

```diff
@@ -0,0 +1,28 @@
+<?php
+
+namespace Drupal\thunder_test_mock_request;
+
+use Drupal\Core\DependencyInjection\ContainerBuilder;
+use Drupal\Core\DependencyInjection\ServiceProviderBase;
+use Symfony\Component\DependencyInjection\Reference;
+
+/**
+ * Add mock handler and HTTP middleware.
+ */
+class ThunderTestMockRequestServiceProvider extends ServiceProviderBase {
+
+  /**
+   * {@inheritdoc}
+   */
+  public function alter(ContainerBuilder $container) {
+    parent::alter($container);
```

Comment:
There really is no need to call the parent. ServiceProviderBase is really odd. I wouldn't use it. I'd use ServiceProviderInterface and be done.